### ISSUE TITLE
collapseChangesAcrossMultipleVersions() no longer available in language services

### ISSUE THREAD

**Issue by derekcicerone** [Microsoft/TypeScript#1709]
Since TypeScript 1.4 has now been released, I'm working on a new develop version of the Eclipse TypeScript plug-in which tracks the master branch of TypeScript (I assume it'll be 1.5) and I noticed that TextChangeRange.collapseChangesAcrossMultipleVersions() is no longer available in typescriptServices.d.ts. This seems to be related to the removal of TextChangeRange and TextSpan.

In general, these classes don't look too difficult to re-create but the collapseChangesAcrossMultipleVersions() function in particular may be challenging. I was wondering if this was an intentional change and if so, would it be possible to expose just the collapseChangesAcrossMultipleVersions() functionality in the new API?

In case you are curious to see where these classes were used in the plug-in codebase, please see:
https://github.com/palantir/eclipse-typescript/blob/master/Bridge/src/fileInfo.ts
and
https://github.com/palantir/eclipse-typescript/blob/master/Bridge/src/snapshot.ts

**Comment by danquirk** [comment]
Tagging @mhegazy @CyrusNajmabadi @vladima who can probably provide the best guidance on this.

**Comment by mhegazy** [comment]
We have unlinked these files from the build in preparation for deleting them. We have not been using the text change range management in a while now, and it was just dead code. We have relied on the host providing this functionality for us and return a single change range for a while now. You are certainly welcome to copy the implementation in textChangeRange and use it; it is under Apache 2.0 licence.
